### PR TITLE
RPi-#115. Add cmd/setup load, and list

### PR DESCRIPTION
**Pull request by YCK1130**
### What is this PR for?

Add cmd/setup load, and list

### What type of PR is it?

 Feature

### Todos

load
-   [X]   dancer pin pairs, name, fps

list
-  [X]  list , dancer pin pairs, name, fps

### What is the Github issue?
 #115

### How should this be tested?

1. read README in command/setup first to build boost/serialization
2. `make all` under command/setup
3. `$ ./load dancer Ken Ken.json`
4. `$ ./list Ken

### Questions:

-   Do the license files need updating? Yes/No
-   Are there breaking changes for older versions? Yes/No
-   Does this need new documentation? Yes/No

### CODE REVIEW

File: 2023/commands/setup/load.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using json = nlohmann::json;
+
+int main(int argc, char *argv[]) {
+    if (argc != 4) {
+        cerr << "Error: missing parameters! (input type, dancer name, json file are needed)" << endl;
+        exit(0); // or 1?
+    }
+
+    string dataType = argv[1];
+    string dancerName = argv[2];
```
Comment (rayray2002) [inline 2023/commands/setup/load.cpp:22]: dancer name could be same as json file name?

File: 2023/commands/setup/load.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    // cout<<jsonFile["OFPARTS"]["Arm_L"]<<endl;
+    if(dataType.compare("dancer")==0){
+        Player dancer;
+        dancer.setDancer(dancerName, jsonFile);
+        cout<<"Date set as below\n";
+        cout<<dancer.list()<<endl;
+
+        string ofileName(boost::archive::tmpdir());
+        ofileName += "/data/"+dancerName+"_DancerData";
```
Comment (rayray2002) [inline 2023/commands/setup/load.cpp:42]: why save to a tmpfile

File: 2023/commands/setup/list.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#include "nlohmann/json.hpp"
+#include "player.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+int main(int argc, char *argv[]) {
+    if (argc != 2) {
+        cerr << "Error: missing parameters! (dancerName is needed)" << endl;
+        exit(0); // or 1?
```
Comment (rayray2002) [inline 2023/commands/setup/list.cpp:17]: 1

File: 2023/commands/setup/list.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    in
+    */
+    string dancerName = argv[1];
+    string fileName(boost::archive::tmpdir());
+    fileName += "/data/"+dancerName+"_DancerData";
+    Player new_dancer;
+    if(!restorePlayer(new_dancer,fileName.c_str())){
+        exit(1);
+    };
+    cout<<dancerName<<"'s date restore as below\n";
```
Comment (rayray2002) [inline 2023/commands/setup/list.cpp:30]: typo

File: 2023/commands/setup/player.cpp [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    for (map<string, LEDStripeSetting >::const_iterator it = LEDPARTS.begin(); it != LEDPARTS.end(); ++it){
+        ostr<<it->first+":{\n    id: "+to_string(it->second.id)+",\n    len: "+to_string(it->second.len)+"\n}\n";
+    }
+    ostr<<"******************************\0";
+    
+    return ostr.str();
+}
+
+template<class Archive>
+void Player::serialize(Archive & ar, const unsigned int version){
```
Comment (rayray2002) [inline 2023/commands/setup/player.cpp:56]: bad variable name

File: 2023/commands/setup/player.cpp [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    ar & dancerName;
+    ar & fps;
+    ar & OFPARTS;
+    ar & LEDPARTS;
+};
+
+
+ostream & operator<<(ostream &os, const Player &player)
```
Comment (rayray2002) [inline 2023/commands/setup/player.cpp:64]: bad variable name

File: 2023/commands/setup/player.cpp [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+{
+    os<<player.list();
+    return os;
+}
+
+void savePlayer(const Player &s, const char * filename){
```
Comment (rayray2002) [inline 2023/commands/setup/player.cpp:70]: bad variable name

File: 2023/commands/setup/player.cpp (hunk 5 above)
Comment (YCK1130) [reply 2023/commands/setup/player.cpp:56]: change ar to archive?

File: 2023/commands/setup/player.cpp (hunk 6 above)
Comment (YCK1130) [reply 2023/commands/setup/player.cpp:64]: os change to ostream?

File: 2023/commands/setup/load.cpp (hunk 2 above)
Comment (YCK1130) [reply 2023/commands/setup/load.cpp:42]: idk... learned from stackoverflow

File: 2023/commands/setup/player.cpp (hunk 5 above)
Comment (rayray2002) [reply 2023/commands/setup/player.cpp]: Ok

File: 2023/commands/setup/player.cpp (hunk 6 above)
Comment (rayray2002) [reply 2023/commands/setup/player.cpp]: Ok

**Review by rayray2002 — APPROVED** [review]

**Pull request merged**